src/test/capi_predict_validation_test.cpp

```diff
@@ -645,6 +660,77 @@ TEST_F(CAPIPredictValidation, RequestIncorrectContentSizeShapeAuto) {
     EXPECT_EQ(status, ovms::StatusCode::INVALID_CONTENT_SIZE) << status.string();
 }
 
+TEST_F(CAPIPredictValidation, ValidRequestWithOutputs) {
+    ovms::signed_shape_t shape = {1, 224, 224, 3};
+    request.addOutput("Output_FP32_1_224_224_3_NHWC", OVMS_DATATYPE_FP32, shape.data(), 4);
+    request.setOutputBuffer("Output_FP32_1_224_224_3_NHWC", outputBuffer.data(), std::accumulate(begin(shape), end(shape), 1.0, std::multiplies<size_t>()) * ovms::DataTypeToByteSize(OVMS_DATATYPE_FP32), OVMS_BUFFERTYPE_CPU, std::nullopt);
+    shape = {1, 3, 62, 62};
+    request.addOutput("Output_U8_1_3_62_62_NCHW", OVMS_DATATYPE_U8, shape.data(), 4);
+    request.setOutputBuffer("Output_U8_1_3_62_62_NCHW", outputBuffer.data(), std::accumulate(begin(shape), end(shape), 1.0, std::multiplies<size_t>()) * ovms::DataTypeToByteSize(OVMS_DATATYPE_U8), OVMS_BUFFERTYPE_CPU, std::nullopt);
+    shape = {1, 6, 128, 128, 16};
+    request.addOutput("Output_U8_1_3_62_62_NCHW", OVMS_DATATYPE_I64, shape.data(), 5);
+    request.setOutputBuffer("Output_U8_1_3_62_62_NCHW", outputBuffer.data(), std::accumulate(begin(shape), end(shape), 1.0, std::multiplies<size_t>()) * ovms::DataTypeToByteSize(OVMS_DATATYPE_I64), OVMS_BUFFERTYPE_CPU, std::nullopt);
+    shape = {1, 2, 8, 4};
+    request.addOutput("Output_U8_1_3_62_62_NCHW", OVMS_DATATYPE_U16, shape.data(), 4);
+    request.setOutputBuffer("Output_U8_1_3_62_62_NCHW", outputBuffer.data(), std::accumulate(begin(shape), end(shape), 1.0, std::multiplies<size_t>()) * ovms::DataTypeToByteSize(OVMS_DATATYPE_U16), OVMS_BUFFERTYPE_CPU, std::nullopt);
+
+    auto status = instance->mockValidate(&request);
+    EXPECT_TRUE(status.ok()) << status.string();
+}
+
+TEST_F(CAPIPredictValidation, OutputWithNoBuffer) {
+    ovms::signed_shape_t shape = {1, 224, 224, 3};
+    request.addOutput("Output_FP32_1_224_224_3_NHWC", OVMS_DATATYPE_FP32, shape.data(), 4);
+
+    auto status = instance->mockValidate(&request);
+    EXPECT_EQ(status, ovms::StatusCode::INVALID_CONTENT_SIZE);
```

Comment:
This is status string:
"Invalid content size of tensor proto"
For INVALID_CONTENT_SIZE. I believe we need special status code for C-API missing buffer (not buffer having wrong size).